src/sentry/search/snuba/executors.py

```diff
@@ -279,38 +278,23 @@ def snuba_search(
 
         bulk_query_results = bulk_raw_query(query_params_for_categories, referrer=referrer)
 
-        # [([row1a, row2a,], totala, row_lengtha), ([row1b, row2b,], totalb, row_lengthb), ...]
-        mapped_results: Sequence[Tuple[Iterable[MergeableRow], int, int]] = list(
-            map(
-                lambda bulk_result: (
-                    bulk_result["data"] if bulk_result["data"] is not None else [],
-                    bulk_result["totals"]["total"]
-                    if bulk_result["totals"]["total"] is not None
-                    else 0,
-                    len(bulk_result),
```

Comment:
I'm pretty sure this value `len(bulk_result)` is completely wrong -- this is the length of a dictionary

it's really difficult to see that in the lefthand side due to the map/filter/reduce obfuscation

I didn't change it in unfolding this as a simple for loop (needed because mypy was unable to trace the types through the indirection) to preserve behaviour

Reply:
Are there any tests covering this at the moment? Would feel better with this change if we have at least something confirming no behavior change.